### ISSUE TITLE
Library not loaded

### ISSUE THREAD

**Issue by AaronHolbrook** [dinodeck/dinodeck#4]
Really excited to start the how to make an rpg, but unfortunately having issues running the initial hello world program.

```
$> ./dinodeck_mac
dyld: Library not loaded: @rpath/libopenal.1.dylib
  Referenced from: /Users/tyrun/dev/games/how-to-make-an-rpg/examples_intro/hello_world/./dinodeck_mac
  Reason: image not found
[1]    76073 trace trap  ./dinodeck_mac
```

**Comment by zencephalon** [comment]
I have the same problem too. Also for compiling dinodeck a list of dependencies would come very in handy. It seems to rely on some libraries I don't have installed.

**Comment by sax** [comment]
I'm running into the same problem. Using `otool -L` I can see that libopenal should be found based on `@rpath`.

```
○ → otool -L dinodeck_mac
dinodeck_mac:
        ...
    @rpath/libopenal.1.dylib (compatibility version 1.0.0, current version 1.17.2)
        ...
```

However when I use `otool -l` there is no `LD_RPATH` listed.

I know this can be fixed with DYLD_LIBRARY_PATH, but I'm looking for a way to fix the binary.

**Comment by sax** [comment]
Another issue is that the binaries are linking to `@rpath/libopenal.1.dylib`, whereas they ship with `libopenal.dylib`. Providing a symlink does not solve the problem, however, because of the lack of `LC_RPATH` in the binary.

**Comment by sax** [comment]
I can solve this by running the following commands for each instance of `dinodeck_mac`

```
install_name_tool -add_rpath "@executable_path/" dinodeck_mac
ln -s libopenal.dylib libopenal.1.dylib
```

**Comment by balaam** [comment]
Wow sax thanks for this! 
I was anticipating a day of messing around with settings and getting access to another mac to try and solve this. I'll update the binaries today.

**Comment by sax** [comment]
Cool! Yeah, there should be a way to pass this info to the linker during compilation, but I don't know how to do it for a Mac.

**Comment by AaronHolbrook** [comment]
Thank you so much @sax !

**Comment by balaam** [comment]
Hi guys,

I updated the binaries in the hello world project this weekend and the book examples. I **also** updated the openal library (libopenal.1.dylib) - the one supplied has a problem but it only occurs when a sound is loaded. So you might want to also update that library - it's available in this [zip](http://static.dinodeck.com/hello_world_mac_1.1.zip).

I'll close this bug, issue once I've verified the fix on a fresh mac which I'll get access to sometime this week.

Thanks!
